conman/routes/managers.py

```diff
@@ -34,6 +36,16 @@ def best_match_for_path(self, path):
             msg = 'No matching Route for URL. (Have you made a root Route?)'
             raise self.model.DoesNotExist(msg)
 
+    def create(self, *, url, **kwargs):
```

Comment:
I would have thought that the defaults (`blank=False, null=False`) for `URLField` would have been enough to stop saving an empty string.

Reply:
Good thinking. I don't think `blank=False` helps with `Route.objects.create()`, as validation is only really ever called from `ModelForm` instances, usually -- `null=False` is a good point though -- the default value appears to be getting set to `''` (blank string) somehow, and I can't see why...

Reply:
Okay, so it looks like [`TextField` defaults to `''`](https://github.com/django/django/blob/1.11.5/django/db/models/fields/__init__.py#L793) because `empty_strings_allowed` is `True` (it's the default on `Field`).

To change this, I'd need to make a new `TextField` subclass, and change that attribute, as django doesn't allow it to be passed as an argument to the constructor.

That wouldn't be crazy, I suppose, but it would mean that we get some kind of `IntegrityError` instead of a `TypeError` when the argument is missed out, and no validation would be performed. Perhaps this is overkill. What do you think?

Reply:
It sounds like keeping it the way it is and keeping it simple is the way to go. Maybe something to keep in mind is problems arise later.